bleak/backends/bluezdbus/utils.py

```diff
@@ -52,7 +52,7 @@ def device_path_from_characteristic_path(characteristic_path: str) -> str:
         A D-Bus object path of the device.
     """
     # /org/bluez/hci1/dev_FA_23_9D_AA_45_46/service000c/char000d
-    return characteristic_path[:37]
+    return "/".join(characteristic_path.split("/")[:5])
```

Comment:
This is in a hot path that gets called a lot, so would want to see that we don't have a performance regression.

Better would probably be:

```suggestion
    return characteristic_path[:-21]
```

Reply:
Thanks for the quick feedback! Performance is a good point, which I didn't think of. Indeed, the `join` version is 5 times slower on my PC:

```
In [1]: N="/org/bluez/hci1/dev_FA_23_9D_AA_45_46/service000c/char000d"

In [2]: %timeit N[:37]
119 ns ± 0.996 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [3]: %timeit "/".join(N.split("/")[:5])
636 ns ± 1.74 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [4]: %timeit N[:-21]
120 ns ± 1.02 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

For my use case, using `[:-21]` would also work. Is it certain that the service and char always have 4 number digits?